### DISCUSSION TITLE
Add package management support for ALT Linux

### DISCUSSION THREAD

**Discussion by Mogaba** [saltstack/salt#67511] [Feature Requests]
Hello, a long time ago Salt was added to ALT Linux: https://github.com/saltstack/salt/issues/3444.

Unfortunately there's still no package management support, the `pkg` module is missing in ALT Linux. Is it possible to implement it?

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html)
  - [Join our Community Slack](https://join.slack.com/t/saltstackcommunity/shared_invite/zt-1zlfxffs1-NuEH~G9TzOeuNGdsfZIl3w)
  - [IRC on LiberaChat](https://web.libera.chat/#salt)
  - [Salt Project YouTube channel](https://www.youtube.com/channel/UCpveTIucFx9ljGelW63-BWg)
  - [Salt Project Twitch channel](https://www.twitch.tv/saltprojectoss)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at saltproject@vmware.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!

**Comment by s0undt3ch** [comment]
@Mogaba the best way to add that kind of support is to have someone who uses the Linux distribution in question contribute the support for it.
I'm adding this to the approved milestone, but that does not mean that someone will be assigned to this any time soon.

**Comment by OrangeDog** [comment]
It appears that ALT Linux uses `rpm` packages, but the `apt` package manager.

Being detected as `Debian` should _mostly_ work, but there's a good chance it will need to be a whole new family.

**Comment by OrangeDog** [comment]
This requires #3473 (currently closed as not planned). It looks like some of the work was already done.

**Comment by OrangeDog** [comment]
How much of it works if you set this in the minion config?
```yaml
providers:
  pkg: aptpkg
  lowpkg: rpm_lowpkg
```

**Comment by nnNzNnn** [comment]
It doesn't work either. 
aptpkg and rpm_lowpkg modules do not support Alt Workstation

> How much of it works if you set this in the minion config?
> 
> ```yaml
> providers:
>   pkg: aptpkg
>   lowpkg: rpm_lowpkg
> ```
